src/Illuminate/Validation/Rules/Dimensions.php

```diff
@@ -113,65 +264,205 @@ public function maxHeight($value)
      */
     public function ratio($value)
     {
-        $this->constraints['ratio'] = $value;
+        $this->ratio = $value;
 
         return $this;
     }
 
     /**
-     * Set the minimum aspect ratio.
+     * Set the minimum aspect ratio constraint.
      *
      * @param  float  $value
      * @return $this
      */
     public function minRatio($value)
     {
-        $this->constraints['min_ratio'] = $value;
+        $this->minRatio = $value;
 
         return $this;
     }
 
     /**
-     * Set the maximum aspect ratio.
+     * Set the maximum aspect ratio constraint.
      *
      * @param  float  $value
      * @return $this
      */
     public function maxRatio($value)
     {
-        $this->constraints['max_ratio'] = $value;
+        $this->maxRatio = $value;
 
         return $this;
     }
 
     /**
-     * Set the aspect ratio range.
+     * Set the aspect ratio between constraint.
      *
      * @param  float  $min
      * @param  float  $max
      * @return $this
      */
     public function ratioBetween($min, $max)
     {
-        $this->constraints['min_ratio'] = $min;
-        $this->constraints['max_ratio'] = $max;
+        $this->ratioBetween = [$min, $max];
 
         return $this;
     }
 
     /**
-     * Convert the rule to a validation string.
+     * Build the array of underlying validation rules based on the current state.
+     *
+     * @return array
+     */
+    protected function buildValidationRules()
+    {
+        $rules = [];
+
+        $validationRules = [
+            'width', 'minWidth', 'maxWidth', 'widthBetween',
+            'height', 'minHeight', 'maxHeight', 'heightBetween',
+            'ratio', 'minRatio', 'maxRatio', 'ratioBetween',
+        ];
+
+        foreach ($validationRules as $property) {
+            if ($this->{$property} !== null) {
+                $value = $this->{$property};
+
+                if (is_array($value)) {
+                    $value = implode(',', $value);
+                }
+
+                $ruleName = Str::snake($property);
+                $rules[] = "{$ruleName}:{$value}";
+            }
+        }
+
+        return array_merge($rules, $this->customRules);
+    }
+
+    /**
+     * Set the default callback to be used for determining default rules.
+     *
+     * If no arguments are passed, the default dimensions rule configuration will be returned.
      *
-     * @return string
+     * @param  static|callable|null  $callback
+     * @return static|void
      */
-    public function __toString()
+    public static function defaults($callback = null)
+    {
+        if (is_null($callback)) {
```

Comment:
You could avoid an unnecessary function call here like this:
```suggestion
        if ($callback === null) {
```